docs/tokens.md

```diff
@@ -3,137 +3,239 @@ id: tokens
+
+### Constructing an ERC721 Token Contract
+
+We'll use ERC721 to track items in our game, which will each have their own unique attributes. Whenever one is to be awarded to a player, it will be minted and sent to them. Players are free to keep their token or trade it with other people as they see fit, as they would any other asset on the blockchain!
+
+Here's what a contract for tokenized items might look like:
 
 ```solidity
-contract DoggoToken is ERC20, ERC20Detailed, ERC20Mintable, ERC20Burnable {
+pragma solidity ^0.5.0;
 
-    constructor(
-        string memory name,
-        string memory symbol,
-        uint8 decimals
-    )
-        ERC20Burnable()
-        ERC20Mintable()
-        ERC20Detailed(name, symbol, decimals)
-        ERC20()
-        public
-    {}
+import "openzeppelin-solidity/contracts/token/ERC721/ERC721Full.sol";
+import "openzeppelin-solidity/contracts/drafts/Counters.sol";
+
+contract GameItem is ERC721Full {
+    using Counters for Counters.Counter;
+    Counters.Counter private _tokenIds;
+
+    constructor() ERC721Full("GameItem", "ITM") public {
+    }
+
+    function awardItem(address player, string memory tokenURI) public returs (uint256) {
+        _tokenIds.increment();
+
+        uint256 newItemId = _tokenIds.current()
+        _mint(player, newItemId);
+        _setTokenURI(newItemId, tokenURI);
+
+        return newItemId;
+    }
 }
 ```
 
-`ERC20Mintable` allows to add minters via `addMinter(addr)`, so they (like the DOGGO Network multisig) can mint tokens to the dogsitters in exchange for watching the nice doggos while their owners leave for vacation. The token is `ERC20Burnable` we want to have the ability to stake DOGGO tokens on our reputation—if the dogsitter does a bad job, their tokens get burned!
+The [`ERC721Full`](api/token/ERC721#erc721full) contract includes all standard extensions, and is probably the one you want to use. In particular, it includes [`ERC721Metadata`](api/token/ERC721#erc721metadata), which provides the [`_setTokenURI`](api/token/ERC721#ERC721Metadata._setTokenURI(uint256,string)) method we use to store an item's metadata.
 
-### A Note on `decimals`
+Also note that, unlike ERC20, ERC721 lacks a `decimals` field, since each token is distinct and cannot be partitioned.
 
-You might remember from the previous chapter about crowdsales about how math is performed in financial situations: **all currency math is done in the smallest unit of that currency**.
+New items can not be created:
 
-That means that the `totalSupply` of a token is actually in what we call `TKNbits`, not what you see as `TKN`. So if my total supply is `1` and we have `5` decimals in the token, that's actually `1 TKNbit` and will be displayed as `0.00001 TKN`.
+```javascript
+> GameItem.awardItem(playerAddress, "https://game.com/item-id-8u5h2m.json")
```

Comment:
These code examples are a bit misleading I think. Because of using the uppercase "class name" `GameItem` instead of an instance like `gameItem`. What do you think?

Reply:
I think either is fine to be honest, and this is pseudocode at best, meant to explain a concept, not show how to do something. Otherwise we'd enter the land of contract objects, etc., which I'd rather avoid here.

Reply:
I would prefer gameItem. Easier to then explain to users later this is an instance.